### PR TITLE
Refactoring summarization class to accept both HTML and text input

### PR DESCRIPTION
**Pull request by T-TonyLiang**
## Changes in this PR

We will be switching over to having the clean text extraction (ie. extracting the article text from HTML) done on the client side. As a result, the backend would need to support a summarization endpoint that accepts text as well as HTML.

- Changes to the server to accept both HTML input and text input from the same endpoint (if the request has `HTML` we will parse the HTML, otherwise we will treat it as clean text.
- Refactoring the text rank class and adding more documentation, splitting the algorithm into smaller functions makes it much easier for unit testing
- More unit tests and fixtures
- Changes to the REPL to be able to send both URL and text requests via. (`:set input text`)
 
There will be a frontend PR out soon that will switch over from passing the HTML to passing clean text.

### CODE REVIEW

File: summarizer_server/tests/test_text_rank.py [hunk 1]
```diff
@@ -1,17 +1,100 @@
...
+            article.title,
+        )
+        self.assertEqual(expected_article_text, article.text)
+        self.assertEqual("en", article.config.get_language())
+
+    def test_summarize_from_html(self):
+        article_html = utils.get_article_contents("article2.html")
+
+        summary = self.text_rank.summarize_from_html(article_html, 15)
+        self.assertTrue(summary)
```
Comment (jerryliu55) [inline summarizer_server/tests/test_text_rank.py]: Is there a reason you assert true? Is it testing for list not empty?

Comment (T-TonyLiang) [reply summarizer_server/tests/test_text_rank.py]: List is not empty and list is not None. Just testing that there are some elements in the list.

File: summarizer_server/tests/test_text_rank.py [hunk 2]
```diff
@@ -1,17 +1,100 @@
...
+
+        self.assertEqual(len(sentences), len(all_top_sentences))
+        self.assertEqual(
+            math.ceil(len(all_top_sentences) * 15 / 100), len(top_15p_sentences)
+        )
+
+    def test_summarize_one_sentence(self):
+        summary = self.text_rank.summarize("Hello world!", "Hello world!", "en", 100)
+
+        self.assertListEqual([], summary)
```
Comment (EssaSaeed) [inline summarizer_server/tests/test_text_rank.py]: Wait, should this be empty, or this this WAI (working as intended)? I would've assumed that 100% of the sentences returned as a summary in english with 1 sentence would return that 1 sentence.

**Comment by T-TonyLiang** [comment]
That seems like a good approach, maybe we could discuss this for the future. One thing that could be a blocker is that we don't load the word vectorizations for our tests locally, although we can just test with the newspaper summarization and drop the text rank.

**Pull request merged**